cpp/log/cert.cc

```diff
@@ -1086,32 +1086,34 @@ Cert::Status TbsCertificate::DerEncoding(string* result) const {
     // is bad until proven otherwise.
     LOG(WARNING) << "Failed to serialize the TBS component";
     LOG_OPENSSL_ERRORS(WARNING);
-    return Cert::FALSE;
+    return util::Status(Code::INTERNAL, "Failed to serialize TBS");
   }
   result->assign(reinterpret_cast<char*>(der_buf), der_length);
   OPENSSL_free(der_buf);
-  return Cert::TRUE;
+  return util::Status::OK;
 }
 
 
-Cert::Status TbsCertificate::DeleteExtension(int extension_nid) {
+util::Status TbsCertificate::DeleteExtension(int extension_nid) {
   if (!IsLoaded()) {
     LOG(ERROR) << "TBS not loaded";
-    return Cert::ERROR;
+    return util::Status(Code::FAILED_PRECONDITION, "Cert not loaded (TBS)");
   }
 
-  int extension_index;
-  Cert::Status status = ExtensionIndex(extension_nid, &extension_index);
-  if (status != Cert::TRUE)
-    return status;
+  StatusOr<int> status = ExtensionIndex(extension_nid);
```

Comment:
Would be better named `extension_index`, I think. Could be made `const`, and use constructor initialisation, while you're there?